core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -80,13 +80,42 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
       new ApiVersions
     )
 
-    new TransactionMarkerSenderThread(threadName, networkClient, brokerListenerName, time)
+    new TransactionMarkerSenderThread(threadName, networkClient, time)
   }
 
+  private case class PendingTxn(transactionMetadata: TransactionMetadata, callback: WriteTxnMarkerCallback)
+
   // use finer synchronization (concurrent map + blocking queue) so that we do not need a global lock
   private val brokerStateMap: concurrent.Map[Int, DestinationBrokerAndQueuedMarkers] = concurrent.TrieMap.empty[Int, DestinationBrokerAndQueuedMarkers]
 
-  private val pendingTxnMap: concurrent.Map[Long, TransactionMetadata] = concurrent.TrieMap.empty[Long, TransactionMetadata]
+  private val pendingTxnMap: concurrent.Map[Long, PendingTxn] = concurrent.TrieMap.empty[Long, PendingTxn]
+
+  // TODO: Config for how often this runs?
```

Comment:
Let's pick a reasonable default for now. If we need to add a config, we can add one later. This is the approach we are following elsewhere as well.